Casks/evernote.rb

```diff
@@ -28,10 +28,11 @@
   livecheck do
     url "https://evernote.s3.amazonaws.com/boron/mac/public/latest-mac.yml"
     strategy :electron_builder do |yml|
-      match = yml["files"][0]["url"].match(/Evernote[._-](\d+(?:\.\d+)+)[._-]mac[._-]ddl[._-]ga[._-](\d+(?:\.\d+)*)/)
+      regex(/Evernote[._-](\d+(?:\.\d+)+)[._-]mac[._-]ddl[._-]ga[._-](\d+(?:\.\d+)*)[._-]([0-9a-fA-F]+)/)
+      match = yml["files"][0]["url"].match(regex)
       next if match.blank?
```

Comment:
```suggestion
      yaml["files"]&.collect do |file|
        match = file["url"]&.match(regex)
        next if match.blank?
```

The existing `livecheck` block uses a looser regex (i.e., excluding the file extension) and naively assumes the first file URL in the YAML file will be appropriate. Instead, it seems like we should be matching `.dmg` files and collecting versions from all the file URLs that match the regex and returning that array from the `strategy` block (allowing livecheck to determine which version is newest).

We're unlikely to end up with more than one version in this particular context but the added flexibility helps to avoid one particular way in which this could fail and require maintenance over time.